stemcell_builder/stages/bosh_go_agent/assets/02-restrict-nats-api-access-allow-monit-api

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# THESE SCRIPTS ARE EXECUTED IN LEXICAL ORDER. MAKE SURE TO CHECK IF ANYTHING ELSE (CURRENTLY MONIT) WRITES OTHER CGROUP BASED RULES TO AVOID INTERFERENCE. IT'S PROBABLY AGOOD IDEA TO -A DROPS AND -I ACCEPTS
+
+source /var/vcap/bosh/etc/nats-access-helper.sh
+
+
+# Give the agent access to Monit right away
+if ! iptables -t mangle -C POSTROUTING -d 127.0.0.1 -p tcp --dport 2822 \
+            -m cgroup --cgroup "${nats_isolation_classid}" -j ACCEPT
+    iptables -t mangle -I POSTROUTING -d 127.0.0.1 -p tcp --dport 2822 \
+             -m cgroup --cgroup "${nats_isolation_classid}" -j ACCEPT
+fi
+
+wait_for_agent_settings() {
+  until [[ -f /var/vcap/bosh/settings.json ]]; do
+    echo 'waiting for /var/vcap/bosh/settings.json';
+    sleep 10;
+  done
+
+  NATS_URL=$(cat /var/vcap/bosh/settings.json | sed 's#.*\("mbus"\):"\(.*\)",.*#\2#g' )
+
+  # CHECK IF THIS IS A CREATE ENV.
+  if grep -q '^https' <<< ${NATS_URL}; then
+    echo "create-env, skipping iptables for nats"
+  else
+    # NATS URL DIDN'T START WITH `https`, PROCEED WITH PARSING OUT VALUE `nats://...@IP:PORT`
+    # GET SECOND FIELD WITH DELIMITER `@` LEAVING `IP:PORT`
+    BOSH_IP_PORT=$( cut -f2 -d@ <<< "${NATS_URL}" )
+    # GET FIRST FIELD WITH DELIMITER `:` LEAVING `IP`
+    BOSH_IP=$( cut -f1 -d: <<< "${BOSH_IP_PORT} )"
+    # GET SECOND FIELD WITH DELIMITER `:` LEAVING `PORT`
+    BOSH_PORT=$( cut -f2 -d: <<< "${BOSH_IP_PORT}" )
+
+    #CHECK IF PARSED OUT IP:PORT COMBINATION IS NOT NIL OR EMPTY
+    if [[ -n ${BOSH_IP_PORT} ]]; then
+      if ! iptables -t mangle -C POSTROUTING -d "${BOSH_IP}" -p tcp --dport "${BOSH_PORT}" \
+                    -m cgroup --cgroup "${nats_isolation_classid}" -j ACCEPT
+      then
+        iptables -t mangle -I POSTROUTING -d "${BOSH_IP}" -p tcp --dport "${BOSH_PORT}" \
+                 -m cgroup --cgroup "${nats_isolation_classid}" -j ACCEPT
+      fi
+      if ! iptables -t mangle -C POSTROUTING -d "${BOSH_IP}" -p tcp --dport "${BOSH_PORT}" \
+                    -m cgroup \! --cgroup "${nats_isolation_classid}" -j DROP
+      then
+        iptables -t mangle -A POSTROUTING -d "${BOSH_IP}" -p tcp --dport "${BOSH_PORT}" \
+                 -m cgroup \! --cgroup "${nats_isolation_classid}" -j DROP
+      fi
+    else
+      echo "no nats protocol connection found. assuming create-env"
```

Comment:
I'm asking because I don't know:

Are we sure that a failure to find the string `@$IP:$PORT` in the NATS URI is a signal that this is a `create-env` Director deploy? In other words: can the NATS URI protocol be anything other than `https` when running in a `create-env`?

If it _cannot_, and we're certain that the `nats` protocol URIs will always have `@$IP:$PORT` at the end then I would rather exit non-zero here, because we're dealing with something unexpected.

Reply:
@klakin-pivotal are you satisfied with the solution in: https://github.com/cloudfoundry/bosh-linux-stemcell-builder/pull/197/commits/63533a42db113ec4d967cef0ca71c86a9860a5f8. If so could you hit approve on this PR?

Reply:
I have pushed the button.